d2l-floating-buttons.html

```diff
@@ -155,7 +195,40 @@
 					innerContainer.style.width = updateWithRect.width + 'px';
 
 				}
+			},
+
+			_tryAdjustSingleRowButtons: function() {
+				var buttons = this.queryAllEffectiveChildren('button');
+				buttons = buttons.filter(function(button) {
+					return (button.offsetHeight > 0);
+				});
+
+				if (buttons.length) {
+					var TOLERANCE = 5,
+						offsetTopValues = [],
+						buttonsShareRow = false,
+						currentOffsetTop,
+						currentOffsetHeight;
+
+					for (var x = 0; x < buttons.length; x++) {
+						currentOffsetTop = buttons[x].offsetTop;
+						for (var y = 0; y < offsetTopValues.length; y++) {
```

Comment:
This would not be the most efficient algorithm for many buttons but since there is usually only a few (can't imagine more than 10) it seems okay to me.